core/src/main/java/lucee/runtime/functions/arrays/ArrayToStruct.java

```diff
@@ -35,20 +35,24 @@
 
 	private static final long serialVersionUID = 2050803318757965798L;
 
-	public static Struct call(PageContext pc, Array arr) throws PageException {
-		Struct sct = new StructImpl();
+	public static Struct call(PageContext pc, Array arr, boolean valueAsKey) throws PageException {
+		Struct sct = new StructImpl(Struct.TYPE_LINKED);
 		int[] keys = arr.intKeys();
 		for (int i = 0; i < keys.length; i++) {
 			int key = keys[i];
-			sct.set(KeyImpl.toKey(key + ""), arr.getE(key));
+			if (valueAsKey)
+				sct.set(KeyImpl.toKey(arr.getE(key) + ""), key);
```

Comment:
here you need to do Caster.toKey(arr.getE(key)) instead, for the int it is fine to do `key+""`, but the value set can be everything, an array a component, whatever and if it is a complex value we need an exception.